### PR TITLE
[IMP] figure: drag & drop detection

### PR DESCRIPTION
**Pull request by hokolomopo**
## Description:

When a figure is drag & dropped , its opacity is set to 0.9.
But the drag & drop starts when we press the mouse button,
and not when we start to move the figure. This means that when we click
a figure to select it, if will become transparent for a brief moment.

We should start the drag & drop state only on `mousemove`,
not on `mousedown`

PR based on #[1737](https://github.com/odoo/o-spreadsheet/pull/1737) to avoid rebases
Odoo task ID : [2951450](https://www.odoo.com/web#id=2951450&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/figures/figure/figure.ts [hunk 1]
```diff
@@ -124,6 +124,7 @@ export class FigureComponent extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-FigureComponent";
   static components = {};
   figureRegistry = figureRegistry;
+  state = useState({ isDragging: false });
```
[the remark this answers is not in the document]
Comment (hokolomopo) [reply src/components/figures/figure/figure.ts]: 🙈

File: src/components/helpers/figure_dnd_manager.ts [hunk 2]
```diff
@@ -43,6 +43,8 @@ export abstract class FigureDndManager {
   protected currentHorizontalSnapLine: HorizontalSnapLine | undefined = undefined;
   protected currentVerticalSnapLine: VerticalSnapLine | undefined = undefined;
 
+  isActive = false;
```
Comment (pro-odoo) [inline src/components/helpers/figure_dnd_manager.ts:46]: protected ?

Comment (hokolomopo) [reply src/components/helpers/figure_dnd_manager.ts:46]: It's used in `figure.xml`

**Comment by hokolomopo** [comment]
Fixed somewhere else

**Pull request closed without merging**